skl2onnx/operator_converters/gaussian_mixture.py

```diff
@@ -0,0 +1,146 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import numpy as np
+from sklearn.mixture.gaussian_mixture import _compute_log_det_cholesky
+from ..common._registration import register_converter
+from ..algebra.onnx_ops import (
+    OnnxAdd, OnnxSub, OnnxMul, OnnxGemm, OnnxReduceSumSquare,
+    OnnxReduceLogSumExp, OnnxExp, OnnxArgMax, OnnxConcat
+)
+
+
+def convert_sklearn_gaussian_mixture(scope, operator, container):
+    """
+    Converter for GaussianMixture.
+
+    Parameters which change the prediction function:
+
+    * *covariance_type*
+    """
+    X = operator.inputs[0]
+    out = operator.outputs
+    op = operator.raw_operator
+    n_features = X.type.shape[1]
+    n_components = op.means_.shape[0]
+
+    # def _estimate_weighted_log_prob(self, X):
+    # self._estimate_log_prob(X) + self._estimate_log_weights()
+    log_weights = np.log(op.weights_)  # self._estimate_log_weights()
+
+    # self._estimate_log_prob(X)
+    log_det = _compute_log_det_cholesky(
+        op.precisions_cholesky_, op.covariance_type, n_features)
+
+    if op.covariance_type == 'full':
+        # shape(op.means_) = (n_components, n_features)
```

Comment:
I see a lot of commented code in this file, could you clean it up?

Reply:
I prefer to let it, it is how it is implemented in scikit-learn, I can add a new comment to specify it comes from sklearn.